### ISSUE TITLE
Finetuning problem: weights become NaN

### ISSUE THREAD

**Issue by agonzgarc** [BVLC/caffe#2315]
I am finetuning a network pre-trained on ILSVRC 2012 for object detection using PASCAL VOC 2012 trainval data. I have kept the original parameter setting proposed, i.e.
base_lr: 0.001
momentum: 0.9
weight_decay: 0.0005

At the beginning everything seems to work fine, the loss is mostly decreasing. After some iterations (usually a few thousand, below 10000 in most cases) the weights start to grow until they become NaNs. This is different from existing issues because in my case, when this happens, the loss becomes a number around 1 and stays there for all the remaining iterations, unlike other issues in which the loss also becomes NaN.

The same finetuning has been replicated in another machine with similar settings but slightly different hardware and the problem doesn't occur. Could this be a problem of my hardware?

Thanks for your help.

**Comment by chengchengowen** [comment]
I also met the same problem. In our case, the parts of the data are random generated.

**Comment by rohrbach** [comment]
Please ask usage questions on [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) -- this issues tracker is primarily for Caffe development discussion. Thanks!
